src/ast/passes/resource_analyser.cpp

```diff
@@ -387,6 +396,31 @@ void ResourceAnalyser::visit(Ternary &ternary)
   }
 }
 
+void ResourceAnalyser::update_variable_info(Variable &var)
+{
+  const auto on_stack_limit = bpftrace_.config_.get(
+      ConfigKeyInt::on_stack_limit);
+  if (var.type.GetSize() > on_stack_limit) {
```

Comment:
This similar logic now seems to be in multiple places in this file. Should we have a helper like `exceedStackLimit(size_t size);`

Reply:
Fixed in latest!